angular/src/components/sso.component.ts

```diff
@@ -161,6 +163,12 @@ export class SsoComponent {
                         },
                     });
                 }
+            } else if (response.forcePasswordReset) {
```

Comment:
We are starting to have very many branches in this function. Which makes it a bit tricky to follow.

One thing that comes to mind that would clean it up a fair bit would be to convert the `on<hook>` to actual methods with a default implementation.

I.e. 

```ts
protected onSuccessfulLoginForceResetNavigate(orgIdFromState) {
    this.router.navigate([this.changePasswordRoute], {
        queryParams: {
            identifier: orgIdFromState,
        },
    });
}
```



 We could simplify things by using proper functions for the `onSuccessfulLoginForceResetNavigate` which by default does `this.router.navigate([this.forcePasswordResetRoute]);`.

We could then look into using guard clauses.

```ts
this.loggingIn = true;
try {
    this.formPromise = this.authService.logInSso(code, codeVerifier, this.redirectUri);
    const response = await this.formPromise;
    
    if (response.twoFactor) {
        this.onSuccessfulLoginTwoFactorNavigate();
        return;
    }

    ...
}
```

Reply:
Noted - will tackle some of this cleanup in another group of PRs.

Reply:
Feel free to make a tech debt item and throw it on Asana if you don't have time atm.